src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -595,6 +595,29 @@ internal bool IsEntryPointCandidate
             get { return IsStatic && Name == WellKnownMemberNames.EntryPointMethodName; }
         }
 
+        internal bool HasAsyncMainReturnType(CSharpCompilation compilation)
```

Comment:
> "HasAsyncMainReturnType" tells me that the return type needs to be ```Task``` or ```Task<int>```

Then the name should covey just that, for example ReturnsTaskOrTaskOfInt, describes exactly what to expect from the function.